CRM/Core/Form.php

```diff
@@ -1666,7 +1666,7 @@ public function addField($name, $props = [], $required = FALSE, $legacyDate = TR
         }
         else {
           $fieldSpec = CRM_Utils_Date::addDateMetadataToField($fieldSpec, $fieldSpec);
-          $attributes = ['format' => $fieldSpec['date_format']];
+          $attributes = ['format' => $fieldSpec['html']['formatType']];
```

Comment:
So it looks like you've done an audit in this PR to fill-in the `formatType` for a long list of fields.

On general principle, it seems like there should be a fallback is in order (e.g. for any extensions which have DAOs that are non-conformant - for any fields that were missed by the audit)?

OTOH, when I grep for `date_format`, I'm having trouble spotting anything that supplies `date_format`. Maybe the old is based on some wonky assumption?

Reply:
date_format is set in the line above - ie addDateMetadataToFields

```
        if (!isset($field['date_format'])) {
          $dateAttributes = CRM_Core_SelectValues::date($fieldMetaData['html']['formatType'], NULL, NULL, NULL, 'Input');
          $field['start_date_years'] = $dateAttributes['minYear'];
          $field['end_date_years'] = $dateAttributes['maxYear'];
          $field['date_format'] = $dateAttributes['format'];
          $field['is_datetime_field'] = TRUE;
          $field['time_format'] = $dateAttributes['time'];
          $field['smarty_view_format'] = $dateAttributes['smarty_view_format'];
        }
```